archive/progress_http/src/main/scala/uk/ac/wellcome/platform/archive/progress_http/Router.scala

```diff
@@ -7,60 +7,96 @@ import akka.http.scaladsl.marshalling.Marshal
   import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
   import uk.ac.wellcome.json.JsonUtil._
   implicit val printer = Printer.noSpaces.copy(dropNullValues = true)
   import uk.ac.wellcome.platform.archive.display.DisplayProvider._
 
-  def routes: Route = {
-    pathPrefix("progress") {
-      post {
-        entity(as[RequestDisplayIngest]) { requestDisplayIngest =>
-          onSuccess(progressStarter.initialise(requestDisplayIngest.toProgress)) {
-            progress =>
-              respondWithHeaders(List(createLocationHeader(progress))) {
-                complete(Created -> ResponseDisplayIngest(progress, contextURL))
-              }
-          }
+  def routes: Route = pathPrefix("progress") {
+    post {
+      entity(as[RequestDisplayIngest]) { requestDisplayIngest =>
+        onSuccess(progressStarter.initialise(requestDisplayIngest.toProgress)) {
+          progress =>
+            respondWithHeaders(List(createLocationHeader(progress))) {
+              complete(Created -> ResponseDisplayIngest(progress, contextURL))
+            }
         }
-
-      } ~ path(JavaUUID) { id: UUID =>
-        get {
-          onSuccess(monitor.get(id)) {
-            case Some(progress) =>
-              complete(ResponseDisplayIngest(progress, contextURL))
-            case None =>
-              complete(NotFound -> "Progress monitor not found!")
-          }
+      }
+    } ~ path(JavaUUID) { id: UUID =>
+      get {
+        onSuccess(progressTracker.get(id)) {
+          case Some(progress) =>
+            complete(ResponseDisplayIngest(progress, contextURL))
+          case None =>
+            complete(NotFound -> "Progress monitor not found!")
         }
       }
+    } ~ path("find-by-bag-id" / Segment) { combinedId: String =>
+      // Temporary route to match colon separated ids '/find-by-bag-id/storageSpace:bagId' used by DLCS
+      // remove when DLCS replaces this by '/find-by-bag-id/storageSpace/bagId'
+      get {
+        val parts = combinedId.split(':')
+        val bagId =
+          BagId(StorageSpace(parts.head), ExternalIdentifier(parts.last))
+        findProgress(bagId)
+      }
+    } ~ path("find-by-bag-id" / Segment / Segment) { (space, id) =>
+      get {
+        val bagId = BagId(StorageSpace(space), ExternalIdentifier(id))
+        findProgress(bagId)
+      }
+    }
+  }
+
+  private def findProgress(bagId: BagId) = {
+    val results = progressTracker.findByBagId(bagId)
+    if (results.nonEmpty && results.forall(_.isRight)) {
+      complete(OK -> results.collect {
+        case Right(bagProgress) => DisplayIngestMinimal(bagProgress)
+      })
+    } else if (results.isEmpty) {
+      complete(NotFound -> List[DisplayIngestMinimal]())
+    } else {
+      complete(
+        InternalServerError -> ErrorResponse(
+          context = contextURL.toString,
+          httpStatus = InternalServerError.intValue,
+          description = results.mkString("\n"),
```

Comment:
minor: will this expose the description externally? Usually for Internal Server Errors we log the description but return a generic response.

Reply:
good point, thanks, changed this.